app/main.py

```diff
@@ -1,3 +1,17 @@
-def shop_trip():
-    # write your code here
-    pass
+from app.customer import Customer
+from app.load_jsons import load_complete_json
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    data = load_complete_json("app/config.json")
+    customers = [Customer(**customer_) for customer_ in data.get("customers")]
+    shops = [Shop(**shop) for shop in data.get("shops")]
+    fuel_price = data.get("FUEL_PRICE")
+
+    for customer in customers:
```

Comment:
You can optimize by removing line 8 ` customers = [Customer(**customer_) for customer_ in data.get("customers")]`
and then you can create customer instances here and call  `.choose_the_cheapest_shop_visit(customer, shops, fuel_price)` on newly created instance